app/lib/cms/system_name.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module CMS::SystemName
+
+  extend ActiveSupport::Concern
+
+  included do
+    validates :system_name, uniqueness: { scope: %i[provider_id type], allow_blank: true },
```

Comment:
While the new format is the one specified here, I think I would keep the whitespace (only ` `, and not others that belong to `\s`), because I think even in the default template (without customer's custom templates) there would be plenty of system names with whitespaces in them, and it would be a problem in production in SaaS.

We might need to check in the production DB, whether there are any other system names that would not match `\A\w[\w\-/_ ]+\z`. I think most should match.

Reply:
I found that there's already a [SystemName module](https://github.com/3scale/porta/blob/7cdee6ccb1fcc983580bd93972f19101ee3e6c50/app/lib/system_name.rb) which purpose is the same than this one I added. I think the best solution is to addapt the CMS models to include that `SystemName` module and remove mine. But that module it's enforcing the format without a space. I suggest either:

1. Allow whitespaces in that `SystemName` format, so for all the models, not only CMS.
2. Create a rake task to fix all system names in CMS to match the new format. Also update the jobs that create the builtin developer portal for new users.
3. Keep the CMS system names not integrated with other models, so we keep this `app/lib/cms/system_name.rb` to enforce another format with whitespaces only for CMS.

I vote for 2.

Reply:
Hm, I'd prefer 3 + "Also update the jobs that create the builtin developer portal for new users."

My reasoning is that I don't want to touch the existing SaaS data. Even though the API was (and still kind of is) unsupported, we don't know if any of the customers actually rely on the system names for their CMS elements. I would avoid breaking it for them

But I guess we need someone else's opinion to break this tie :wink: 
@akostadinov ?

Reply:
If system names are used in content to refer to them, then Daria's suggestion makes more sense.

But Joan, if you desire very much consistency, you probably can grep somehow through the portals to check for that, then we can change names. Just be sure that this grepping doesn't kill MySQL or another part of system. And I guess it can take a lot of time. We can discuss this if you want to take on it.

Reply:
It's ok, I'll go with @mayorova's approach

Reply:
After taking a closer look at this, there's no need to update the jobs that create the dev. portal for new users. All builtin content is created sending parameters to the existing models. Since the models now have the new `SystemName ` module included, all `system_names` are generated in the new format. So @mayorova's approach is already implemented.